### ISSUE TITLE
`accounts --unused` lists unused accounts with child accounts

### ISSUE THREAD

**Issue by DamienCassou** [simonmichael/hledger#1971]
If I have the following ledger file:

```
account asset
account asset:current
account asset:current:cash

account expense
account expense:food

2022/12/12 misc
    asset:current:cash                           -10
    expense:food                                  10
```

And try to list unused accounts, I get all non-leaf accounts:

```
$ hledger -f foo.hledger account --unused
asset
asset:current
expense
```

I was expecting an empty output as **I** consider "asset", "asset:current" and "expense" to be used. To me, these non-leaf accounts are "used" because they have "used" child accounts.

A workaround is to stop declaring the non-leaf accounts as I never put any money in them (only in the leaf accounts). If I remember correctly, declaring non-leaf accounts helped me get the accounts sorted properly in reports but I'm not sure anymore.

**Comment by simonmichael** [comment]
Yes, you need to declare parent accounts if you want to alter their display order.

As a workaround, how about using account tags to distinguish leaf and non-leaf accounts ? I've used a `postable:` tag inspired by  https://github.com/simonmichael/hledger/blob/master/bin/hledger-check-postable.hs:

```
account asset
account asset:current
account asset:current:cash  ; postable:

account expense
account expense:food  ; postable:

2022/12/12 misc
    asset:current:cash                           -10
    expense:food                                  10
```
```
$ hledger accounts tag:postable --unused
```

**Comment by tanelihuuskonen** [comment]
If you tag explicitly all your postable accounts, you can also find postings inadvertently made to unpostable parent accounts like this:
`hledger accounts --used "not:tag:postable"`

**Comment by DamienCassou** [comment]
Thank you both @simonmichael and @tanelihuuskonen.

I consider my issue solved so you may close the issue. You may also let it open to get an easier solution implemented.
